PSKoans/Koans/IntroTraining/AboutGetMember.koans.ps1

```diff
@@ -0,0 +1,137 @@
+using module PSKoans
+[Koan(Position = 106)]
+param()
+<#
+
+    Introduction to Get-Member
+ 
+    Following on from our last Koan about cmdlet verbs, lets cover the nouns.
+
+    As previously stated, the noun portion indicates the target of a given command.
+
+    If you use the "Get-Member" cmdlet on any object you'll see its type name, or data type.
+
+    For example:
+
+        PS C:\Users\Administrator\Documents\GitHub\Dev\PSKoans> Get-Process | Get-Member
+
+
+            TypeName: System.Diagnostics.Process
+
+        Name                       MemberType     Definition
+        ----                       ----------     ----------
+        Handles                    AliasProperty  Handles = Handlecount
+        Name                       AliasProperty  Name = ProcessName
+        NPM                        AliasProperty  NPM = NonpagedSystemMemorySize64
+        PM                         AliasProperty  PM = PagedMemorySize64
+        SI                         AliasProperty  SI = SessionId
+        VM                         AliasProperty  VM = VirtualMemorySize64
+        WS                         AliasProperty  WS = WorkingSet64
+
+    You'll also see properties and methods. Together with the TypeName, this tells us everything
+    we need to know in order to access the object's data and any actions it can perform.
+
+    Accessing a property or method is done via dot notation, which boils down to:
+
+    $Object.Property
+    $Object.Method()
+
+    We'll cover why we have to use parentheses with method calls later.
+
+#>
+
+Describe "Get Member" {
+<#
+    Use Get-Member to identify properties or methods of some common objects!
+
+    Make sure you use a different cmdlet each time, we'll be checking :)
+#>
+
+    It 'allows us to explore properties on an object' {
+
+        # Get some properties!
+        
+        <# EXAMPLE
+
+        $cmdlet1 = 'Get-Process'
+        $PropertyName = "Threads"
+        $Reason = $BecauseString -f $PropertyName, $cmdlet1
+        Get-Process |
+            Get-Member -MemberType Property -Name $PropertyName |
+            Should -Not -BeNullOrEmpty -Because $Reason
+
+        #>
+        $BecauseString = "property '{0}' should be present in output from {1}"
+
+        $cmdlet1 = '__'
+        $PropertyName = "__"
+        $Reason = $BecauseString -f $PropertyName, $cmdlet1
+        ____ |
+            Get-Member -MemberType Property -Name $PropertyName |
+            Should -Not -BeNullOrEmpty -Because $Reason
```

Comment:
I think this is going to end up being a bit too confusing with the double blank for the cmdlet. Maybe something like this could work a little better? 🙂 

```powershell
$Cmdlet1 = '____'
$PropertyName = '____'

$Reason = $BecauseString -f $PropertyName, $cmdlet1
& (Get-Command -Name $Cmdlet1) |
    Get-Member -MemberType Property -Name $PropertyName |
    Should -Not -BeNullOrEmpty -Because $Reason
```

Reply:
🦄